### PR TITLE
add neW community page

### LINKED ISSUE #34: New Community Page Basic Layout
- contains text box for community name
- contains text area for community description
- able to upload image for community background
- able to see the image that was uploaded and delete it if necessary
- create a new community button in the communities dropdown in the navbar that links to this page
image

### PR DESCRIPTION
**Pull request by Xiaoqi510**
# Description
This implements the new community creation page which

- contains text box for community name
- contains text area for community description
- able to upload image for community background
- able to see the image that was uploaded and delete it if necessary
- create a new community button in the communities dropdown in the navbar that links to this page

## Related Issue

Solves #34

## Type of change

- [x] New feature (enhancement)

## How Has This Been Tested?

- [x] Manual testing

# Checklist:

- [x] Does a similar (open or closed) pull request not already exist?
- [x] Is the pull request _head_ repository a fork repository?
- [x] Is the pull request _compare_ branch a development branch?
- [x] Is the code documented, particularly in hard-to-understand areas?
- [x] Does the code build without new warnings?
- [ ] Has testing been performed that proves changes are effective and work?
- [x] Has a self- and/or peer-review of the code been performed?
- [x] Have dependent changes been merged and published in downstream modules?
- [ ] Does all new and existing automated testing pass?
- [x] Is the person responsible for the repository assigned to the pull request?
- [x] Is the pull request linked to a project?
- [x] Is the pull request linked to a milestone?

For more information, refer to the [Contributing Guidelines](../blob/contribution-guidelines/.github/CONTRIBUTING.md) document.

### CODE REVIEW

**Review by christopher-alba — CHANGES_REQUESTED** [review]
image
Heya please fix this warning so we can see the Netlify preview :) Thanks!

**Comment by Xiaoqi510** [comment]
> image Heya please fix this warning so we can see the Netlify preview :) Thanks!

i fixed it

File: src/pages/newCommunity/NewCommunity.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  }, [selectedFile]);
+
+  const onSelectFile = (e) => {
+    if (!e.target.files || e.target.files.length === 0) {
+      setSelectedFile(undefined);
+      return;
+    }
+
+    setSelectedFile(e.target.files[0]);
+  };
```
Comment (christopher-alba) [inline src/pages/newCommunity/NewCommunity.jsx]: Following a discussion with @Xiaoqi510 a change has been requested to convert the image to a base64 string before storing it into a variable, which requires additional code in this area. This is to allow for the image to be stored efficiently in the backend.

Comment (Xiaoqi510) [reply src/pages/newCommunity/NewCommunity.jsx]: okay, fixed it. thank you for your advice

File: src/pages/newCommunity/newCommunity.css [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+.ncDescription {
+  margin-left: 2%;
+}
+.ncUploadButton {
+  margin-left: 4%;
+}
+.ncSubmitButton {
+  margin-right: 4%;
+  float: right;
+}
```
Comment (christopher-alba) [inline src/pages/newCommunity/newCommunity.css:32]:
```suggestion
  .ncCarouselWrapper{
    border-radius: 4px;
    margin-left: 2%;
    margin-right: 2%;
    overflow: hidden;
  }
}
```

File: src/pages/newCommunity/NewCommunity.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+                maxRows={15}
+                minRows={15}
+                id="outlined-basic"
+                label="Community Description"
+                variant="outlined"
+              />
+            </div>
+          </Grid>
+          <Grid item xs={12}>
+            <div className="npCarouselWrapper">
```
Comment (christopher-alba) [inline src/pages/newCommunity/NewCommunity.jsx]:
```suggestion
            <div className="ncCarouselWrapper">
```

**Review by christopher-alba — CHANGES_REQUESTED** [review]
Hey its looking great! Just have one visual bug regarding the carousel. 
image
I made some suggestions on how to fix it. Thanks

File: src/pages/newCommunity/NewCommunity.jsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+          <Grid item xs={6}>
+            <div className="ncSubmitButton">
+              <Button variant="contained">Create</Button>
+            </div>
+          </Grid>
+        </Grid>
+      </form>
+
+      <div></div>
```
Comment (hpar461) [inline src/pages/newCommunity/NewCommunity.jsx]: There seems to be a redundant div tag here; if it's not for a specific purpose, I think it should be removed.

```suggestion
```

**Review by hpar461 — CHANGES_REQUESTED** [review]
Other than the issues that was mentioned by Chris and the one I just reviewed, LGTM!

**Review by hpar461 — DISMISSED** [review]

File: src/pages/newCommunity/newCommunity.css (hunk 2 above)
Comment (Xiaoqi510) [reply src/pages/newCommunity/newCommunity.css:32]: okay, thank you for your advice, fixed the bug

File: src/pages/newCommunity/NewCommunity.jsx (hunk 4 above)
Comment (Xiaoqi510) [reply src/pages/newCommunity/NewCommunity.jsx]: thank you for your suggestion, i deleted that

File: src/pages/newCommunity/NewCommunity.jsx (hunk 3 above)
Comment (Xiaoqi510) [reply src/pages/newCommunity/NewCommunity.jsx]: changed that, ty

**Review by christopher-alba — APPROVED** [review]
LGTM!

**Review by yuchendong13 — APPROVED** [review]
Looks good, nice job!

**Pull request merged**